Branch-SDK/src/main/java/io/branch/referral/util/BranchEvent.java

```diff
@@ -256,6 +269,175 @@ public boolean logEvent(Context context) {
+                                                        .setPrice(price, currency)
+                                                        .setProductName(product.getName())
+                                                        .setQuantity((double) quantity)
+                                                        .setContentSchema(BranchContentSchema.COMMERCE_PRODUCT));
+                                buos.add(buo);
+                            }
+
+                            new BranchEvent(BRANCH_STANDARD_EVENT.PURCHASE)
+                                    .setCurrency(currency)
+                                    .setDescription(purchase.getOrderId())
+                                    .setCustomerEventAlias(purchasedProductType)
+                                    .setRevenue(revenue)
+                                    .addCustomDataProperty("package_name", purchase.getPackageName())
+                                    .addCustomDataProperty("order_id", purchase.getOrderId())
+                                    .addCustomDataProperty("loggedFromIAP", "true")
+                                    .addCustomDataProperty("is_auto_renewing", String.valueOf(purchase.isAutoRenewing()))
+                                    .addCustomDataProperty("purchase_token", purchase.getPurchaseToken())
+                                    .addContentItems(buos)
+                                    .logEvent(context);
+                        } else {
+                            Log.e("BranchEvent", "Failed to query product details. Error code: " + billingResult.getResponseCode());
+                        }
+                    });
+
+
+                    billingClient.queryProductDetailsAsync(queryProductDetailsParams, (billingQueryResult, productDetailsList) -> {
+                        if (billingQueryResult.getResponseCode() == BillingClient.BillingResponseCode.OK) {
+                            List<BranchUniversalObject> buos = new ArrayList<>();
+                            CurrencyType currency = null;
+                            double revenue = 0.00;
+                            String purchasedProductType = "";
+
+                            for (ProductDetails product : productDetailsList) {
+                                double price;
+                                int quantity;
+
+                                if (product.getProductType() == "inapp_subscription") {
+                                    purchasedProductType = "Subscription";
+                                    currency = CurrencyType.valueOf(product.getSubscriptionOfferDetails().get(0).getPricingPhases().getPricingPhaseList().get(0).getPriceCurrencyCode());
+                                    price = product.getSubscriptionOfferDetails().get(0).getPricingPhases().getPricingPhaseList().get(0).getPriceAmountMicros() / 1_000_000;
+                                    quantity = 1;
+                                    revenue += price * quantity;
```

Comment:
sort of nit: this line is identical for both if/else, you can move it below the if block and delete the two redundant lines